pages/open-library/book/[id].tsx

```diff
@@ -0,0 +1,456 @@
+                            <h5 className="mb-0">{review.reviewer}</h5>
+                            <div>
+                              {[...Array(5)].map((_, i) => (
+                                <span key={i} className="text-warning">
+                                  {i < review.rating ? '\u2605' : '\u2606'}
+                                </span>
+                              ))}
+                            </div>
+                          </div>
+                          <Card.Text>{review.comment}</Card.Text>
+                          <Card.Text>
+                            <small className="text-muted">
+                              {new Date(review.date).toLocaleDateString()}
+                            </small>
+                          </Card.Text>
+                        </Card.Body>
+                      </Card>
+                    ))}
+                    <div className="text-center mt-4">
+                      <Button
+                        variant="outline-primary"
+                        href="https://open-source-bazaar.feishu.cn/share/base/form/shrcnFAX8q7mnywCQOsVnIS5Dwb"
+                        target="_blank"
+                        rel="noopener noreferrer"
+                      >
+                        Add Your Review
+                      </Button>
+                    </div>
+                  </div>
+                ) : (
+                  <div className="text-center py-5">
+                    <p className="mb-4">
+                      No reviews yet. Be the first to review this book!
+                    </p>
+                    <Button
+                      variant="primary"
+                      href="https://open-source-bazaar.feishu.cn/share/base/form/shrcnFAX8q7mnywCQOsVnIS5Dwb"
+                      target="_blank"
+                      rel="noopener noreferrer"
+                    >
+                      Write a Review
+                    </Button>
+                  </div>
+                )}
+              </Tab>
+              <Tab eventKey="history" title="Borrow History">
+                {book.borrowHistory && book.borrowHistory.length > 0 ? (
+                  <div className="table-responsive">
+                    <table className="table table-hover">
+                      <thead>
+                        <tr>
+                          <th>Borrower</th>
+                          <th>Borrow Date</th>
+                          <th>Return Date</th>
+                          <th>Status</th>
+                        </tr>
+                      </thead>
+                      <tbody>
+                        {book.borrowHistory.map((history, index) => (
+                          <tr key={index}>
+                            <td>{history.borrower}</td>
+                            <td>
+                              {new Date(
+                                history.borrowDate,
+                              ).toLocaleDateString()}
```

Comment:
```suggestion
                              {formatDate(history.borrowDate, 'YYYY-MM-DD')}
```
1. 服务端地区设置和用户浏览器通常不同，`toLocaleXXX()` 的结果也会不同，会出现 SSR 报错
2. https://web-cell.dev/web-utility/functions/formatDate.html